core/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -353,6 +370,7 @@ public void writeToThin(StreamOutput out) throws IOException {
         }
 
         out.writeBoolean(primary);
+        out.writeVLong(primaryTerm);
```

Comment:
This serializes the same information that we already have in indexmetadata  (num_replicas+1) times in the cluster state.